optimize/precondition.py

```diff
@@ -0,0 +1,33 @@
+import numpy as np
+from ff.handlers.bonded import HarmonicAngleHandler, HarmonicBondHandler, ImproperTorsionHandler, ProperTorsionHandler
+from ff.handlers.nonbonded import LennardJonesHandler, AM1CCCHandler
+
+default_learning_rates = {
+    HarmonicBondHandler:    np.zeros(2),    # k, length
+    HarmonicAngleHandler:   np.zeros(2),    # k, angle
+    ProperTorsionHandler:   np.zeros(3),    # k, phase, periodicity
+    ImproperTorsionHandler: np.zeros(3),    # k, phase, periodicity
+    AM1CCCHandler:          np.ones(1),     # charge increment
+    LennardJonesHandler:  np.array([0, 1]), # sigma, epsilon
+}
+
+def learning_rates_like_params(
+        ordered_handles, ordered_params, learning_rates=default_learning_rates
+):
+    """Get a list of arrays of same shape as ordered_params, but containing
+    learning rates"""
+
+    ordered_learning_rates = []
+    for handle, params in zip(ordered_handles, ordered_params):
+        key = handle.__class__
+        lr_row = learning_rates[key]
+        lr_array = np.array([lr_row] * len(params))
+
+        if key == AM1CCCHandler:
```

Comment:
Would having AM1CCCHandler learning_rates be a scalar, i.e. `1.0`, as opposed to a `np.ones(1)` solve the shape mismatch?

Reply:
That should also do it -- addressed in https://github.com/proteneer/timemachine/pull/490/commits/f61b242c786b4c53544d57ff8ced21e836bbc66a